crates/modelardb_server/src/parser.rs

```diff
@@ -86,6 +86,16 @@ impl ModelarDbDialect {
         parser.expect_keyword(Keyword::TABLE)?;
         let table_name = self.parse_word_value(parser)?;
 
+        // Check that the table name is not a restricted keyword.
+        for keyword in ALL_KEYWORDS {
+            if table_name == keyword.to_uppercase() || table_name == keyword.to_lowercase() {
```

Comment:
I still think the call to `to_uppercase()` should be moved outside the loop like in my comment above so it is not executed for each iteration of the loop. Also it is unclear why it is necessary to allocate a string for each keyword instead of comparing `&str`s with `&table_name_uppercase == keyword`.